motionplan/motionPlanner.go

```diff
@@ -86,24 +86,49 @@ func (mp *planner) checkPath(planOpts *PlannerOptions, seedInputs, target []fram
 	return ok
 }
 
-// needed to wrap slices so we can use them as map keys.
-type node struct {
+// Node interface is used to wrap a configuration for planning purposes.
+type Node interface {
+	// return the configuration associated with the node
+	Q() []frame.Input
+}
+
+type basicNode struct {
 	q []frame.Input
-	// TODO(rb): do not add extra fields to this parameter, if the need arises
+}
+
+func (n *basicNode) Q() []frame.Input {
+	return n.q
+}
+
+type costNode struct {
+	Node
 	cost float64
 }
 
-type nodePair struct{ a, b *node }
+func newCostNode(q []frame.Input, cost float64) *costNode {
+	return &costNode{&basicNode{q: q}, cost}
+}
+
+type nodePair struct{ a, b Node }
+
+func (np *nodePair) sumCosts() float64 {
```

Comment:
Actually, this is something I'd rather push off until we need it, since the only reason this struct exists anyway is to group nodes that are shared in two maps, and there are no algorithms (or plans for more) that will be using more than 2 trees.  It will be easy to fix later if we need to.

Reply:
Fine with this, just add a TODO